pkg/frontend/openshiftcluster_putorpatch_test.go

```diff
@@ -667,8 +680,9 @@ func TestPutOrPatchOpenShiftCluster(t *testing.T) {
 			asyncOperations := mock_database.NewMockAsyncOperations(controller)
 			openShiftClusters := mock_database.NewMockOpenShiftClusters(controller)
 			subscriptions := mock_database.NewMockSubscriptions(controller)
+			enricher := mock_clusterdata.NewMockOpenShiftClusterEnricher(controller)
```

Comment:
I think we need a test case where the enricher has a side effect on doc, i.e. using https://godoc.org/github.com/golang/mock/gomock#Call.Do.  https://github.com/Azure/ARO-RP/issues/414.